### PR TITLE
view required signoffs in the rule update page

### PR DESCRIPTION
**Pull request by AMUZY**
@bhearsum @gabrielBusta  In an attempt to solve the issue  about [showing signoffs required in rule update page](https://github.com/mozilla-releng/balrog/issues/1147) :  I came up with solutions below : 
Note that the main file edited is the ui/src/views/Rules/Rule/index.jsx. most other files were edited by my vscode extension (prettier)

Main changes I made to the ui/src/views/Rules/Rule/index.jsx
```
 const [requiredSignoffs, fetchRequiredSignoffs] =
    useAction(getRequiredSignoffs);

  useEffect(() => {
    let myPromise = new Promise((res) => {
      res(fetchRequiredSignoffs(OBJECT_NAMES.PRODUCT_REQUIRED_SIGNOFF));
    });
    myPromise.then((rs) => {
      setAllrequiredSignoffs(rs.data.data.required_signoffs);
    });
  }, []);

  // SignOff style
  const signoffStyle = {
    div: {
      display: "flex",
      flexFlow: "row wrap",
      width: "100%",
      margin: "8px 0 8px 0",
      alignItems: "center",
    },
    p1: {
      marginRight: "8px",
      color: "gray",
    },
    p2: {
      color: "black",
    },
  };

  return (
    <Dashboard title={getTitle()}>
      {isLoading && <Spinner loading />}
      {error && <ErrorPanel fixed error={error} />}
      {!isLoading && (
        /*
        THE SIGN OFF DETAIL SHOULD GO JUST UNDERNEATH THE OPENING FRAGMENT
        It should be in the form : 
        "SignOffs Required : 1 signoff from a releng user" 
        */
        <Fragment>
          {/* SHOWS REQUIRED SIGNOFFS */}
          <div style={signoffStyle.div}>
            <Typography component="p" variant="body2" style={signoffStyle.p1}>Required Signoff(s) :</Typography>
            {allRequiredSignOffs.map((reqSignOff, index, all) => {
              if (
                rule.product === reqSignOff.product &&
                rule.channel === `${reqSignOff.channel}*`
              ) {
                return (
                  <Typography component="p" variant="body2" key={index} style={signoffStyle.p2}>{`${
                    reqSignOff.signoffs_required
                  } member${reqSignOff.count > 1 ? "s" : ""} of ${
                    reqSignOff.role
                  }${all[index + 1] ? "," : ""}`}</Typography>
                );
              } else {
                return (
                  <Typography component="p" variant="body2" key={index} style={signoffStyle.p2}>
                    No signoffs provided for this product in local development
                  </Typography>
                );
              }
            })}
            {/* END OF REQUIRED SIGNOFFS INFO */}
          </div>
```

release

So, from my understanding about what @gabrielBusta explained in another issue, the only available signoff available in local development is the "1 member of releng" for the Firefox product (precisely in the release chanel). 
So for products which do not have required signoffs in local development, we would have : 
none
Note that : For products with multiple required signoffs, all signoffs required would be displayed in a comma separated list.

### CODE REVIEW

File: ui/src/views/Rules/Rule/index.jsx
Comment (bhearsum) [inline ui/src/views/Rules/Rule/index.jsx:462]: I'm going to let @gabrielBusta do the full review of this, but I do want to make sure that we don't hardcode a string like this. Even in production there are some products/channels that do not require signoffs. A simple string of "None" would be just fine here.

Comment (AMUZY) [reply ui/src/views/Rules/Rule/index.jsx]: Noted. Changed this. I'm still trying to fix lint errors in my code though. `yarn lint --fix` is giving me errors still. I hope I fix this early enough

Comment (AMUZY) [reply ui/src/views/Rules/Rule/index.jsx]: Done fixing eslint errors now :)

Comment (AMUZY) [reply ui/src/views/Rules/Rule/index.jsx]: Just added a fix for the release channel . Meant to be "release" and not "release*" :)

Comment (AMUZY) [reply ui/src/views/Rules/Rule/index.jsx]: In addition, this also works when creating rules, as soon as you select it's product name and channel, it'll immediately show you the required signoffs right there! :)

**Review by gabrielBusta — CHANGES_REQUESTED** [review]
Great work @AMUZY! This is almost there.

I see "None" displayed in the UI. Can you just display the required sign-offs?

Example of a rule that requires 1 sign-offs from releng:

Screenshot 2023-10-17 at 18 10 07

Example of a rule that requires sign-offs from releng and relman:

Screenshot 2023-10-17 at 18 09 58

File: ui/package.json
Comment (AMUZY) [reply ui/package.json:7]: I'll work on this ASAP !

File: ui/src/views/Rules/Rule/index.jsx
Comment (AMUZY) [reply ui/src/views/Rules/Rule/index.jsx:409]: I'll work on this too.

Comment (AMUZY) [reply ui/src/views/Rules/Rule/index.jsx]: All corrections have been made. Please test now :)

File: ui/package.json
Comment (AMUZY) [reply ui/package.json]: Done

**Comment by bhearsum** [comment]
@AMUZY - FYI: when this is ready for ready again please make sure to explicitly flag someone with the "Re-request review" button near the top right. (If you don't have permission to use that, just flag me and/or @gabrielBusta in a comment instead.)

**Comment by AMUZY** [comment]
Ooooh! thank you! Please it's been ready  siiiince last week. I've just re requested review now.

**Review by bhearsum — CHANGES_REQUESTED** [review]
It looks like this doesn't account for globs (`*`) in channel names. For example, https://localhost:9000/rules/269 reports no signoffs will be required even though they will, but https://localhost:9000/rules/145 correctly reports the required signoffs.

Other changes may be needed as well, but this is what immediately stands out.

**Comment by AMUZY** [comment]
It's done now whoosh! thank you @bhearsum @gabrielBusta for all the support so far. Feels so cool working with y'all. Let me know if there are further changes needed please.

**Comment by AMUZY** [comment]
I hope the code I've written so far is simple enough for easy understanding? Please do let me know if they need some more work so as to help me serve better code blocks 😊

File: ui/src/views/Rules/Rule/index.jsx
Comment (AMUZY) [reply ui/src/views/Rules/Rule/index.jsx]: Hi @bhearsum I've used it now lol

**Comment by AMUZY** [comment]
Or... hang on a bit...

**Comment by AMUZY** [comment]
Hi @bhearsum , after having a deeper look, I found out that the `ruleMatchesChannel` function doesn't work here still.
it causes a situation whereby it doesn't consider the product name, rather only the channel name.
So therefore it causes signoff to be required on a product that isn't suppose to require signoff for updates :
wrong

FirefoxVPN isn't meant to require signoffs, but signoffs are required simply because the `release` channel is selected

**Comment by AMUZY** [comment]
My previous fix works so I'm gonna undo this commit

**Comment by AMUZY** [comment]
Hi @bhearsum , the `ruleMatchesChannel` function doesn't work here. It causes a situation whereby signoffs are required from rule updates that aren't meant to require signoffs :

wrong

product `FirefoxVPN` isn't meant to require signoffs but it is, simply because the `release` channel is selected

**Comment by AMUZY** [comment]
So I've reversed those changes. My previous fix works fine

**Comment by bhearsum** [comment]
This ended up getting fixed in #3025. If you'd like to finish your work here for experience that's fine - or you can feel free to close this and focus elsewhere.

**Comment by AMUZY** [comment]
alright alright no problem thanks

**Comment by AMUZY** [comment]
Hi @bhearsum @gabrielBusta good day, this PR has been ready for another review for a whle now. Please kindly take a look whenever you have the time thanks

**Comment by bhearsum** [comment]
> Hi @bhearsum @gabrielBusta good day, this PR has been ready for another review for a whle now. Please kindly take a look whenever you have the time thanks

This looks OK, and definitely shows the signoffs. There are some minor issues such as trailing commas, eg:
image

In any case, #3025 already fixed this issue, so I'll leave it up to you whether you want to continue with your own implementation.

**Comment by AMUZY** [comment]
OMG the trailing comma! Oh well...

**Pull request closed without merging**
